pmm/pmm3-submodules.groovy

```diff
@@ -0,0 +1,506 @@
+                    sh '''
+                        set -o errexit
+                        aws ecr-public get-login-password --region us-east-1 | docker login -u AWS --password-stdin public.ecr.aws/e7j3v3n0
+
+                        ${PATH_TO_SCRIPTS}/build-client-srpm public.ecr.aws/e7j3v3n0/rpmbuild:ol9
+                    '''
+                }
+            }
+        }
+        stage('Build client binary rpm EL7') {
+            when {
+                 beforeAgent true
+                 allOf{
+                     expression{ params.BUILD_OS == "el7" }
+                     expression{ env.PMM_VER =~ '^3.' }
+                     }
+            }
+            steps {
+                withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: 'AMI/OVF', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
+                    sh """
+                        set -o errexit
+
+                        aws ecr-public get-login-password --region us-east-1 | docker login -u AWS --password-stdin public.ecr.aws/e7j3v3n0
+
+                        ${PATH_TO_SCRIPTS}/build-client-rpm centos:7
+
+                        mkdir -p tmp/pmm-server/RPMS/
+                        cp results/rpm/pmm-client-*.rpm tmp/pmm-server/RPMS/
+                    """
+                }
+            }
+        }
+        stage('Build client binary rpm') {
+            when {
+                 beforeAgent true
+                 allOf{
+                     expression{ params.BUILD_OS == "el9" }
+                     expression{ env.PMM_VER =~ '^3.' }
+                     }
+            }
+            steps {
+                withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: 'AMI/OVF', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
+                    sh '''
+                        set -o errexit
+
+                        aws ecr-public get-login-password --region us-east-1 | docker login -u AWS --password-stdin public.ecr.aws/e7j3v3n0
+
+                        ${PATH_TO_SCRIPTS}/build-client-rpm public.ecr.aws/e7j3v3n0/rpmbuild:ol9
+
+                        mkdir -p tmp/pmm-server/RPMS/
+                        cp results/rpm/pmm-client-*.rpm tmp/pmm-server/RPMS/
+                    '''
+                }
+            }
+        }
+        stage('Build client docker') {
+            when {
+                        beforeAgent true
+                        expression {
+                            env.PMM_VER =~ '^3.'
+                        }
+                  }
+            steps {
+                withCredentials([usernamePassword(credentialsId: 'hub.docker.com', passwordVariable: 'PASS', usernameVariable: 'USER')]) {
+                    sh """
```

Comment:
This is super unsafe. Please change to single quotes.

Reply:
@ademidoff I copied the pmm2-submodules script and made the changes accordingly for pmm3. So, should this be changed for pmm2 scripts as well?

Reply:
I have no doubt it should, but probably as a separate effort. I have refactored a lot of double quotes in the past, be we need to refactor all of them. It's worth noticing, that it is only critical whenever the secrets appear inside double quotes.

Read more here https://www.jenkins.io/doc/book/pipeline/jenkinsfile/#injection-via-interpolation